client/src/pages/Settings/Availability/Availability.tsx

```diff
@@ -0,0 +1,311 @@
+                tuesday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+                wednesday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+                thursday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+                friday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+                saturday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+                sunday: {
+                  active: initialDayValues.active,
+                  startTime: initialDayValues.startTime,
+                  endTime: initialDayValues.endTime,
+                },
+              }}
+              onSubmit={handleNewSubmit}
+            >
+              {({ values, setFieldValue, handleChange, handleSubmit, isSubmitting }) => (
+                <form onSubmit={handleSubmit}>
+                  <Box display="flex" flexDirection="row" alignItems="center" height="100%">
+                    <FormInput
+                      className={classes.scheduleNameInput}
+                      id="scheduleSelect"
+                      label="New Schedule"
+                      margin="dense"
+                      name="name"
+                      placeholder="Schedule name"
+                      autoComplete="name"
+                      autoFocus
+                      value={values.name}
+                      onChange={handleChange}
+                    />
+                    {showActive && <StarIcon></StarIcon>}
+                  </Box>
+                  {daysOfTheWeek.map((day) => {
+                    return (
+                      <DayOfWeekInput
+                        values={values}
+                        setFieldValue={setFieldValue}
+                        key={day}
+                        day={day}
+                      ></DayOfWeekInput>
+                    );
+                  })}
+                  <Box display="flex" flexDirection="row" justifyContent="center" height="100%">
+                    {!newScheduleCreated && (
+                      <Button type="submit" className={classes.saveButton}>
+                        Save Schedule
+                      </Button>
+                    )}
+                    {newScheduleCreated && !showActive && (
+                      <Button onClick={() => handleMakeActiveSchedule(newScheduleId)}>
```

Comment:
Same as comment above

Reply:
Dont we have to define this one as an anonymous function because we are passing a parameter? If we do onClick={function(parameter)} it will fire the callback function. I know in classes you could bind the parameter, but not sure how to avoid this with functional components.

Reply:
@natey37 You are right, I must've failed to notice the parameter, you can leave it as is - no change necessary